flybywiresim/a32nx
# fix(mcdu): fix negative efob on flight plan page

Fixes #7449 
Fixes #6798 
Fixes #4688
## Summary of Changes
When the flight plan page is active in the mcdu the EFOB will not be updated. This is caused by only updating the `_routeTripFuelWeight` when the fuel pred page is active. Also the MLW will not being updated on the EFB with correct dest efob for Payload Calculations, With this change the `tryUpdateRouteTrip` will be added to the flight plan page aswell. 

## Screenshots (if necessary)



## References





## Additional context



Discord username (if different from GitHub):
Revyn112#1010

## Testing instructions

1. Activate DevTools
2. Open Tools -> Behaviors
3. Switch to LocalVariables and monitor "A32NX_DESTINATION_FUEL_ON_BOARD" 
4. Init and setup your flight like you always do
5. After FUEL PRED Page is setup and F-PLN / FUEL PRED page is active the "A32NX_DESTINATION_FUEL_ON_BOARD" should be updated as soon as there is fuel consumption (APU/ENGINE).
6. As soon as you're in flight and the F-PLN / FUEL PRED page is active the EFOB should match the "A32NX_DESTINATION_FUEL_ON_BOARD" value.
(The EFOB will not change when still on ground).


## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page
